### ISSUE TITLE
re-introduce `ghcup compile cabal`

### ISSUE THREAD

**Issue by hasufell** [haskell/ghcup-hs#264]
In GitLab by @maerwald on Oct 2, 2021, 04:45

**Comment by hasufell** [comment]
@Hecate any news from cabal folks?

**Comment by hasufell** [comment]
@Kleidukos any news from cabal folks?

**Comment by Mikolaj** [comment]
@hasufell: hi!

I tried to ask on gitlab, but the ticket was blobked: what bootstrap script has been removed?

**Comment by hasufell** [comment]
> @hasufell: hi!
> 
> I tried to ask on gitlab, but the ticket was blobked: what bootstrap script has been removed?

It was a shell script and @phadej back then made clear it's not for end users, but for development purposes. It got rewritten in Python or so and I removed that feature.

**Comment by Mikolaj** [comment]
Oh, I see. Indeed we warn "utility is only intended for use in building cabal-install on a new platform". But your use case seems legit. Do the instructions at

https://github.com/haskell/cabal/tree/master/bootstrap

work for you?

**Comment by hasufell** [comment]
> Oh, I see. Indeed we warn "utility is only intended for use in building cabal-install on a new platform". But your use case seems legit. Do the instructions at
> 
> https://github.com/haskell/cabal/tree/master/bootstrap
> 
> work for you?

No, because that only works for the pre-defined GHCs in that directory.

**Comment by Mikolaj** [comment]
> No, because that only works for the pre-defined GHCs in that directory.

Oh, of course. So we'd need to maintain this for all GHCs (I guess major versions should be enough and a couple of minor versions where base version changed? but probably it's simplest to cover all minor versions systematically), which is beyond our capacity currently. I'm afraid I don't see a good solution until somebody volunteers to maintain the bootstrap scripts specifically.

**Comment by hasufell** [comment]
> > No, because that only works for the pre-defined GHCs in that directory.
> 
> I'm afraid I don't see a good solution until somebody volunteers to maintain the bootstrap scripts specifically.

I don't have capacity to maintain this.

**Comment by Mikolaj** [comment]
> I don't have capacity to maintain this.

I wasn't even suggesting that. Your time is particularly precious, given that you maintain such an important tool for the Haskell ecosystem.
